### ISSUE TITLE
Rename the editor "Scene" dock to "SceneTree"

### ISSUE THREAD

**Issue by marcospb19** [godotengine/godot-proposals#846]
**Describe the project you are working on:**
I have a small personal projects, it's a 2d arcade space shooter.

**Describe the problem or limitation you are having in your project:**
The naming of `Scene Dock` to refer to the Scene Tree is misleading mainly counterintuitive.

See the engine UI:

![image](https://user-images.githubusercontent.com/38900226/84442209-8bead300-ac13-11ea-8011-00687d4e6fe0.png)

If I had to explain the editor to someone new, I would say that we have scene tree at the left, and the scene contents at the right.

At the left we have the hierarchy of the nodes in the tree, at the right we have how the scene is arranged, what is the position of the sprites, where the spaceship guns are.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
I suggest that the `Scene` dock should be renamed to `Scene Tree` to better indicate what it represents.

`Scene Tree` is a more intuitive name to this dock, the file associated with this dock is called `scene_tree_dock.cpp` already!!!!

(This change requires editing documentation appearances to match it.)

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
According to the simplicity of it, I don't think it's necessary.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
It cannot be worked around with a few lines of script.

**Is there a reason why this should be core and not an add-on in the asset library?:**
It cannot be implemented as an addon.

**Comment by Calinou** [comment]
In the interest of making high-quality proposals, please [read the rules](https://github.com/godotengine/godot-proposals#readme) and fill in the issue template. This proposal will be reopened once this is done :slightly_smiling_face:

**Comment by marcospb19** [comment]
Hey, @Calinou, thanks for your response.

> This proposal will be reopened once this is done 🙂

Right now I don't have a nice and elaborated justification, so I'll just leave this untouched for a bit, and maybe later resume the discussion (if I find a nice reason to).

**Comment by marcospb19** [comment]
Consider this image from a college project I develop last year with @erneani.

![image](https://user-images.githubusercontent.com/38900226/84442209-8bead300-ac13-11ea-8011-00687d4e6fe0.png)

If I were to explain the editor to someone, I would tell that we have a scene appearing at the right of the image, where we can see all it's content, how it is arranged, and you also have the tabs (at the top) of each open scene you want to switch to (or close).

This display at the left shows how the tree is organized, the hierarchy, you can see which nodes are children of who, and this is indeed the tab's **main function**, you _cannot_ think about it without imagining a `tree`!.

On the other side, the display on the right isn't associated with a tree at all, it is a level, a building, a vehicle, and so on.

---

In conclusion, following this thinking, I think that `Scene` should be renamed to `Scene Tree` to better indicate what it represents.

In addition to this I'll also suggest that the display at the right of the image (center of the engine), continues to be referred as `Viewport`, but that we should call it's content a `Scene` (when there is content displayed) without having the name to conflict with the new renamed `Scene Tree` dock **anymore**, And the tabs above the `Viewport`, `Scene Tabs`.

**Comment by marcospb19** [comment]
It is worth mentioning that the files for this Dock are called `scene_tree_dock.cpp` and `scene_tree_dock.h`, so they are already called `Scene Tree Dock` officially in some places (as expected).

**Comment by Calinou** [comment]
@marcospb19 If you want your propsoal to be reopened, please edit the first post to follow the [issue template](https://github.com/godotengine/godot-proposals/blob/master/.github/ISSUE_TEMPLATE/feature---enhancement-proposal.md).

**Comment by marcospb19** [comment]
Done, sorry for my ignorance, completing the template for a simple proposal made it verbose without adding useful information.
I'll make sure to strictly follow the template on next reports.

**Comment by timothyqiu** [comment]
I prefer to call this a scene instead of scene tree to avoid confusion.

There's a `SceneTree` class, which is an implementation of `MainLoop`. Its instance is _the_ tree that can be entered/exited.

**Comment by YuriSizov** [comment]
> It cannot be implemented as an addon.

It can, actually. But I doubt people would install an addon just for that.
